### PR TITLE
Better testing

### LINKED ISSUE #11: Write more and better tests
**Is your feature request related to a problem? Please describe.**
I'd like to request the addition of test cases for the `Deck` and `Card` classes to ensure their functionality and reliability in the project.

**Describe the solution you'd like**
I'd like to have a comprehensive set of test cases written for the `Deck` and `Card` classes. These test cases should cover various aspects of the classes, including constructor behavior, method functionality, edge cases, and immutability checks. Writing tests will help ensure the correctness and robustness of these classes, which is essential for maintaining code quality.

**Describe alternatives you've considered**
While writing test cases for these classes is the preferred solution, an alternative could be to rely on manual testing. However, this approach is less reliable and may lead to undetected issues in the future. Automated tests provide a systematic and repeatable way to verify the behavior of these classes.

**Additional context**
Adding unit tests for the `Deck` and `Card` classes will not only ensure that the classes work as intended but also make future maintenance and enhancements more manageable. The tests should cover a wide range of scenarios, including both typical and edge cases, to minimize the risk of regressions and ensure long-term stability.

### PR DESCRIPTION
**Pull request by LarsGast**
Closes #11.

### CODE REVIEW

File: DeckOfPlayingCardsLibrary/Card.cs [hunk 1]
```diff
@@ -59,17 +59,6 @@ public Card(Rank rank, Suit suit)
...
-        /// Two cards are equal if they have the same rank and the same suit.
-        /// </summary>
-        /// <param name="other">The other card to compare.</param>
-        /// <returns>true if the cards are equal; otherwise, false.</returns>
-        public bool Equals(Card other)
-        {
-            return this.Rank == other.Rank && this.Suit == other.Suit;
-        }
-
         /// <summary>
```
[the remark this answers is not in the document]
Comment (LarsGast) [reply DeckOfPlayingCardsLibrary/Card.cs:62]: Outside the scope of this PR.

File: DeckOfPlayingCardsTests/Unit/DeckTests.cs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            }
+
+            var after = deck.Draw();
+
+            // Assert
+            Assert.Multiple(() =>
+            {
+                Assert.That(drawn.All(c => c is not null), Is.True, "Expected 52 non-null draws.");
+                Assert.That(
+                    drawn.Select(c => Key(c!)).Distinct().Count(),
```
[the remark this answers is not in the document]
Comment (LarsGast) [reply DeckOfPlayingCardsTests/Unit/DeckTests.cs:60]: The line before checks for nullability.

File: DeckOfPlayingCardsTests/Unit/DeckTests.cs [hunk 3]
```diff
@@ -0,0 +1,250 @@
+﻿namespace DeckOfPlayingCardsTests.Unit
+{
+    [TestFixture]
+    public class DeckTests
+    {
```
[the remark this answers is not in the document]
Comment (LarsGast) [reply DeckOfPlayingCardsTests/Unit/DeckTests.cs:5]: Might remove the only method left in the `Card` class, making it a simple record.

File: DeckOfPlayingCardsTests/Unit/DeckTests.cs [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            _ = deck.Draw();
+            _ = deck.Draw();
+            deck.Reset();
+
+            var firstAfterRest = deck.Draw()!;
+
+            // Assert
+            Assert.That(
+                Key(firstBeforeReset),
+                Is.EqualTo(Key(firstAfterRest)),
```
[the remark this answers is not in the document]
Comment (LarsGast) [reply DeckOfPlayingCardsTests/Unit/DeckTests.cs]: Fixed.
